### PR TITLE
update(userspace): Make HAVE_STRLCPY internal.

### PR DESCRIPTION
**Pull request by geraldcombs**
Create common_config.h, and define HAVE_STRLCPY there. This removes the need for external projects to define HAVE_STRLCPY by hand.

Signed-off-by: Gerald Combs <gerald@wireshark.org>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

/area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

When I build a project that links with libscap+libsinsp here on macOS I have to define HAVE_STRLCPY by hand. This removes that requirement.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -76,11 +76,13 @@ check_symbol_exists(strlcpy "string.h" HAVE_STRLCPY)
 
 if(HAVE_STRLCPY)
 	message(STATUS "Existing strlcpy found, will *not* use local definition by setting -DHAVE_STRLCPY.")
-	add_definitions(-DHAVE_STRLCPY)
 else()
 	message(STATUS "No strlcpy found, will use local definition")
 endif()
 
+configure_file(userspace/common/common_config.h.in ${PROJECT_BINARY_DIR}/common/common_config.h)
```
Comment (Molter73) [inline CMakeLists.txt:83]: I like the approach of moving the generated header file into the `${PROJECT_BINARY_DIR}/` directory, is should make it much easier to cleanup, even if it does lead to some extra configuration needed in cmake.

**Review by Molter73 — DISMISSED** [review]
/lgtm

**Comment by Molter73** [comment]
/hold
Just because I'm not sure we are out of the freeze yet

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**